browser/ui/webui/brave_wallet/wallet_panel_ui_browsertest.cc

```diff
@@ -5,66 +5,296 @@
 
 #include <string>
 
-#include "base/test/scoped_feature_list.h"
+#include "base/strings/string_util.h"
+#include "base/strings/stringprintf.h"
+#include "base/test/bind.h"
+#include "brave/browser/brave_wallet/json_rpc_service_factory.h"
+#include "brave/browser/brave_wallet/keyring_service_factory.h"
+#include "brave/browser/ui/webui/brave_settings_ui.h"
 #include "brave/browser/ui/webui/brave_wallet/wallet_panel_ui.h"
+#include "brave/components/brave_wallet/browser/brave_wallet_utils.h"
+#include "brave/components/brave_wallet/browser/json_rpc_service.h"
+#include "brave/components/brave_wallet/browser/keyring_service.h"
 #include "brave/components/brave_wallet/common/features.h"
 #include "brave/components/constants/webui_url_constants.h"
 #include "chrome/browser/profiles/profile.h"
 #include "chrome/browser/ui/browser.h"
 #include "chrome/browser/ui/browser_tabstrip.h"
 #include "chrome/common/chrome_isolated_world_ids.h"
 #include "chrome/test/base/in_process_browser_test.h"
+#include "chrome/test/base/ui_test_utils.h"
+#include "components/javascript_dialogs/app_modal_dialog_controller.h"
+#include "components/javascript_dialogs/app_modal_dialog_view.h"
+#include "components/javascript_dialogs/tab_modal_dialog_manager.h"
+#include "components/web_modal/web_contents_modal_dialog_manager.h"
 #include "content/public/test/browser_test.h"
 #include "content/public/test/test_utils.h"
+#include "services/network/public/cpp/weak_wrapper_shared_url_loader_factory.h"
+#include "services/network/test/test_url_loader_factory.h"
 
-class WalletPanelUIBrowserTest : public InProcessBrowserTest {
- public:
-  void SetUp() override {
-    feature_list_.InitAndEnableFeature(
-        brave_wallet::features::kNativeBraveWalletFeature);
-    InProcessBrowserTest::SetUp();
+using content::EvalJsResult;
+
+namespace {
+
+std::string SelectInNetworkList(const std::string& selector) {
+  return base::StringPrintf("window.testing.walletNetworks.querySelector(`%s`)",
+                            selector.c_str());
+}
+
+std::string SelectInAddNetworkDialog(const std::string& selector) {
+  return base::StringPrintf(
+      "window.testing.addWalletNetworkDialog.querySelector(`%s`)",
+      selector.c_str());
+}
+
+std::string DoubleClick() {
+  return ".dispatchEvent((function (){const e = "
+         "document.createEvent('MouseEvents');e.initEvent('dblclick',true,true)"
+         ";return e;})())";
+}
+
+std::string CeloNetwork() {
+  return "[data-test-chain-id='chain-0xa4ec']";
+}
+
+std::string PolygonNetwork() {
+  return "[data-test-chain-id='chain-0x89']";
+}
+
+std::string NetworkNameSpan() {
+  return "[class|='NetworkName']";
+}
+
+std::string CeloNetworkHideButton() {
+  return CeloNetwork() + " .hide-network-button";
+}
+
+std::string CeloNetworkChainName() {
+  return CeloNetwork() + " .chainName";
+}
+
+std::string NetworksButton() {
+  return R"([data-test-id='select-network-button'])";
+}
+
+std::string Select(const std::string& selector) {
```

Comment:
nit: `Select` seems ambiguous, like we are changing state to make something selected. These functions are returning JS strings that perform a query for an element. How about `FindElementJS` or `QuerySelectorJS`?

Reply:
`QuerySelectorJS` https://github.com/brave/brave-core/pull/13792/commits/817d5acee8a7a47295a98e651f189797e801f010